### ISSUE TITLE
Release 0.12 - Testing required: Ubuntu 16.04

### ISSUE THREAD

**Issue by reaperzn** [rackerlabs/auter#178]
This issue has been raised as part of the release process for Version 0.12

Please could auter be fully tested on the OS mentioned in the title of this issue. 

# Notes:
 - Please assign yourself to the issue **ONLY IF** you are actively working on it
 - Please ensure you are testing the code from the **v0.12 branch** of this repo
 - Follow the testing guide from https://github.com/rackerlabs/auter/blob/develop/HACKING.md
 - Any problems should be documented in this issue
 - If there are changes required, after the issue has been validated, a new issue should be raised and linked referenced in this issue

**Comment by popindavibe** [comment]
### Manual building of unsigned package
```
builder@xenial:~/auter/auter-0.12$ debuild -us -uc
 dpkg-buildpackage -rfakeroot -D -us -uc
dpkg-buildpackage: source package auter
dpkg-buildpackage: source version 0.12
dpkg-buildpackage: source distribution xenial
dpkg-buildpackage: source changed by Paolo Gigante <paolo.gigante.sa@gmail.com>
 dpkg-source --before-build auter-0.12
dpkg-buildpackage: host architecture amd64
 fakeroot debian/rules clean
dh clean
   dh_testdir
   dh_auto_clean
        make -j1 clean
make[1]: Entering directory '/home/builder/auter/auter-0.12'
make[1]: Leaving directory '/home/builder/auter/auter-0.12'
   dh_clean
 dpkg-source -b auter-0.12
dpkg-source: warning: no source format specified in debian/source/format, see dpkg-source(1)
dpkg-source: info: using source format '1.0'
dpkg-source: info: building auter using existing auter_0.12.orig.tar.gz
dpkg-source: info: building auter in auter_0.12.diff.gz
dpkg-source: warning: diff '/home/builder/auter/auter_0.12.diff.gz.new.7cx6Uz' doesn't contain any patch
dpkg-source: info: building auter in auter_0.12.dsc
 debian/rules build
dh build
   dh_testdir
   dh_update_autotools_config
   dh_auto_configure
   dh_auto_build
        make -j1
make[1]: Entering directory '/home/builder/auter/auter-0.12'
make[1]: Leaving directory '/home/builder/auter/auter-0.12'
   dh_auto_test
 fakeroot debian/rules binary
dh binary
   dh_testroot
   dh_prep
   dh_installdirs
   dh_auto_install
   dh_install
   dh_installdocs
   dh_installchangelogs
   dh_installman
   dh_installcron
   dh_perl
   dh_link
   dh_strip_nondeterminism
   dh_compress
   dh_fixperms
   dh_installdeb
   dh_gencontrol
dpkg-gencontrol: warning: Depends field of package auter: unknown substitution variable ${shlibs:Depends}
   dh_md5sums
   dh_builddeb
dpkg-deb: building package 'auter' in '../auter_0.12_all.deb'.
 dpkg-genchanges  >../auter_0.12_amd64.changes
dpkg-genchanges: including full source code in upload
 dpkg-source --after-build auter-0.12
dpkg-buildpackage: full upload (original source is included)
Now running lintian...
W: auter source: non-native-package-with-native-version
W: auter source: empty-debian-diff
W: auter: debian-changelog-line-too-long line 7
W: auter: debian-changelog-line-too-long line 9
W: auter: debian-changelog-line-too-long line 10
W: auter: manpage-has-errors-from-man usr/share/man/man1/auter.1.gz 49: warning [p 1, 9.5i]: cannot adjust line
W: auter: script-not-executable usr/lib/auter/auter.module
Finished running lintian.
```

### Install Deb
```
root@xenial:/home/builder/auter# dpkg -i auter_0.12_all.deb
Selecting previously unselected package auter.
(Reading database ... 34758 files and directories currently installed.)
Preparing to unpack auter_0.12_all.deb ...
Unpacking auter (0.12) ...
Setting up auter (0.12) ...
INFO: auter enabled
Processing triggers for man-db (2.7.5-1) ...
```
### Basic testing
Using:
```
root@xenial:~# cat auter-testing.sh
#!/bin/bash

function test_options() {

  for i in disable status enable status help version; do
    echo "auter --${i}"
    auter --${i}
    echo "exit code is: $?"
  done

}

### MAIN ###
test_options

exit 0
```

```
root@xenial:~# ./auter-testing.sh
auter --disable
INFO: auter disabled
exit code is: 0
auter --status
auter is currently disabled
exit code is: 0
auter --enable
INFO: auter enabled
exit code is: 0
auter --status
auter is currently enabled and not running
exit code is: 0
auter --help

Usage: auter [--enable|--disable|--status] [--prep] [--apply] [--reboot] [--postreboot] [OPTIONS]

Automatic Update Transaction Execution by Rackspace. A wrapper around cron and yum/dnf/apt to manage system updates with the ability to configure automatic reboots and custom scripts.

Actions:
  --enable      Enable auter
  --disable     Disable auter. Also deletes unused pidfile if it exists
  --status      Show whether enabled or disabled
  --prep        Pre-download updates before applying
  --apply       Apply updates, and reboot if AUTOREBOOT=yes
  --reboot      Reboot system including pre/post reboot scripts
  --postreboot  Run post reboot script

Options:
  --config=FILE Specify the full path to an auter config file. Defaults to /etc/auter/auter.conf
  --stdout      Always log to STDOUT, regardless of not having a tty
  --skip-all-scripts
                Skip the executions of all custom scripts (Default in /etc/auter/*.d/)

  --skip-scripts-by-phase=PHASE
                Skip the execution of the custom scripts for the specified phase. You can specify myltiple phases.

                Valid Phases: pre-prep, post-prep, pre-apply, post-apply, pre-reboot, post-reboot.

                Example: --skip-scripts-by-phase="pre-prep,post-apply,pre-reboot"

  --skip-scripts-by-name=SCRIPTNAME
                Skip specific scripts by name. You can specify myltiple phases.

                Example: --skip-scripts-by-name="10-configsnap-pre, 20-startApp.sh"

  --no-wall     If possible, suppress shutdown wall messages in the reboot phase
  -h, --help    Show this help text
  -v, --version Show the version

exit code is: 0
auter --version
auter 0.12
exit code is: 0
```

- Man page OK
- Could not get the signed package to build

**Comment by popindavibe** [comment]
## Scenario 1
- pre-download YES
- onlyinstallfromprep NO

### Testing Prep
```
root@xenial:/home/builder/auter# egrep -v '^$|^#' /etc/auter/auter.conf /etc/cron.d/auter
/etc/auter/auter.conf:CONFIGSET="default"
/etc/auter/auter.conf:AUTOREBOOT="no"
/etc/auter/auter.conf:PACKAGEMANAGEROPTIONS=""
/etc/auter/auter.conf:PREDOWNLOADUPDATES="yes"
/etc/auter/auter.conf:ONLYINSTALLFROMPREP="no"
/etc/auter/auter.conf:MAXDELAY="3600"
/etc/auter/auter.conf:PACKAGEMANAGERLOCKWAITTIME="60"
/etc/auter/auter.conf:PACKAGEMANAGERLOCKRETRIES="5"
/etc/auter/auter.conf:PREPREPSCRIPTDIR="/etc/auter/pre-prep.d"
/etc/auter/auter.conf:POSTPREPSCRIPTDIR="/etc/auter/post-prep.d"
/etc/auter/auter.conf:PREAPPLYSCRIPTDIR="/etc/auter/pre-apply.d"
/etc/auter/auter.conf:POSTAPPLYSCRIPTDIR="/etc/auter/post-apply.d"
/etc/auter/auter.conf:PREREBOOTSCRIPTDIR="/etc/auter/pre-reboot.d"
/etc/auter/auter.conf:POSTREBOOTSCRIPTDIR="/etc/auter/post-reboot.d"
root@xenial:/home/builder/auter# auter --prep
INFO: Running with: /usr/sbin/auter --prep
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Running Pre-Prep script /etc/auter/pre-prep.d/pre_prep_script
INFO: Updates downloaded
INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
```

Testing with ksh package:
```
root@xenial:~# apt-cache policy ksh
ksh:
  Installed: 93u+20120801-2
  Candidate: 93u+20120801-2ubuntu0.16.04.1
  Version table:
     93u+20120801-2ubuntu0.16.04.1 500
        500 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages
 *** 93u+20120801-2 500
        500 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages
        100 /var/lib/dpkg/status
root@xenial:~# ls -l /var/cache/apt/archives/ksh_93u+20120801-2*
-rw-r--r-- 1 root root 821788 Jul  6  2017 /var/cache/apt/archives/ksh_93u+20120801-2ubuntu0.16.04.1_amd64.deb
```

### Testing Apply 
```
root@xenial:/home/builder/auter# auter --apply
INFO: Running with: /usr/sbin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
INFO: Applying updates
Trying to update
INFO: Updates complete. You may need to reboot for some updates to take effect
INFO: Auter successfully ran at 2018-07-19T18:48:04+00:00
INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script

root@xenial:/home/builder/auter# cat /var/lib/auter/last-apply-output-default
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following package was automatically installed and is no longer required:
  libfreetype6
Use 'apt autoremove' to remove it.
The following packages will be upgraded:
  ksh
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/822 kB of archives.
After this operation, 1024 B of additional disk space will be used.
(Reading database ... 34818 files and directories currently installed.)
Preparing to unpack .../ksh_93u+20120801-2ubuntu0.16.04.1_amd64.deb ...
Unpacking ksh (93u+20120801-2ubuntu0.16.04.1) over (93u+20120801-2) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up ksh (93u+20120801-2ubuntu0.16.04.1) ...
update-binfmts: warning: unable to open /proc/sys/fs/binfmt_misc/status for writing: Permission denied
update-binfmts: warning: unable to open /proc/sys/fs/binfmt_misc/register for writing: Permission denied
update-binfmts: warning: unable to enable binary format ksh


root@xenial:~# apt-cache policy ksh
ksh:
  Installed: 93u+20120801-2ubuntu0.16.04.1
  Candidate: 93u+20120801-2ubuntu0.16.04.1
  Version table:
 *** 93u+20120801-2ubuntu0.16.04.1 500
        500 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages
        100 /var/lib/dpkg/status
     93u+20120801-2 500
        500 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages
```
### Logs output
```
Jul 19 18:42:54 xenial root[30568]: custom pre prep script ran
Jul 19 18:43:18 xenial auter[30812]: INFO: Updates downloaded
Jul 19 18:43:18 xenial auter[30822]: INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
Jul 19 18:43:18 xenial root[30824]: custom post prep script ran
Jul 19 18:48:02 xenial auter[30861]: INFO: Running with: /usr/sbin/auter --apply
Jul 19 18:48:02 xenial auter[30865]: INFO: Running in an interactive shell, disabling all random sleeps
Jul 19 18:48:02 xenial auter[30878]: INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
Jul 19 18:48:02 xenial root[30880]: custom pre apply script ran
Jul 19 18:48:02 xenial auter[30884]: INFO: Applying updates
Jul 19 18:48:04 xenial auter[31231]: INFO: Updates complete. You may need to reboot for some updates to take effect
Jul 19 18:48:04 xenial auter[31235]: INFO: Auter successfully ran at 2018-07-19T18:48:04+00:00
Jul 19 18:48:04 xenial auter[31238]: INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script
Jul 19 18:48:04 xenial root[31240]: custom post apply script ran
```

**Comment by popindavibe** [comment]
## Scenario 2
- pre-download YES
- onlyinstallfromprep YES

### Testing Prep
```
root@xenial:/home/builder/auter# sed -r -i 's/(ONLYINSTALLFROMPREP=).*/\1"yes"/g' /etc/auter/auter.conf
root@xenial:/home/builder/auter# apt-get clean
root@xenial:/home/builder/auter# auter --prep
INFO: Running with: /usr/sbin/auter --prep
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Running Pre-Prep script /etc/auter/pre-prep.d/pre_prep_script
INFO: Updates downloaded to /var/cache/auter/default
INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
root@xenial:/home/builder/auter# ls -lh /var/cache/auter/
total 0
drwxr-xr-x 1 root root 108 Jul 19 19:00 default
root@xenial:/home/builder/auter# ls -lh /var/cache/auter/default/
total 804K
-rw-r--r-- 1 root root 803K Jul  6  2017 ksh_93u+20120801-2ubuntu0.16.04.1_amd64.deb
-rw-r----- 1 root root    0 Jul 19 19:00 lock
drwx------ 1 _apt root    0 Jul 19 19:00 partial
```

### Testing Apply
```
root@xenial:/home/builder/auter# auter --apply
INFO: Running with: /usr/sbin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
INFO: Applying updates
Trying to update
INFO: Updates complete. You may need to reboot for some updates to take effect
INFO: Auter successfully ran at 2018-07-19T19:02:16+00:00
INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script

root@xenial:~# apt-cache policy ksh
ksh:
  Installed: 93u+20120801-2ubuntu0.16.04.1
  Candidate: 93u+20120801-2ubuntu0.16.04.1
  Version table:
 *** 93u+20120801-2ubuntu0.16.04.1 500
        500 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages
        100 /var/lib/dpkg/status
     93u+20120801-2 500
        500 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages
```

### Logs
```
Jul 19 19:00:17 xenial auter[31731]: INFO: Running with: /usr/sbin/auter --prep
Jul 19 19:00:17 xenial auter[31735]: INFO: Running in an interactive shell, disabling all random sleeps
Jul 19 19:00:17 xenial auter[31738]: INFO: Running Pre-Prep script /etc/auter/pre-prep.d/pre_prep_script
Jul 19 19:00:17 xenial root[31740]: custom pre prep script ran
Jul 19 19:00:30 xenial auter[31987]: INFO: Updates downloaded to /var/cache/auter/default
Jul 19 19:00:30 xenial auter[31998]: INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
Jul 19 19:00:30 xenial root[32000]: custom post prep script ran
Jul 19 19:02:11 xenial auter[32014]: INFO: Running with: /usr/sbin/auter --apply
Jul 19 19:02:11 xenial auter[32018]: INFO: Running in an interactive shell, disabling all random sleeps
Jul 19 19:02:12 xenial auter[32038]: INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
Jul 19 19:02:12 xenial root[32040]: custom pre apply script ran
Jul 19 19:02:12 xenial auter[32044]: INFO: Applying updates
Jul 19 19:02:16 xenial auter[32401]: INFO: Updates complete. You may need to reboot for some updates to take effect
Jul 19 19:02:16 xenial auter[32405]: INFO: Auter successfully ran at 2018-07-19T19:02:16+00:00
Jul 19 19:02:16 xenial auter[32408]: INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script
Jul 19 19:02:16 xenial root[32410]: custom post apply script ran
```

**Comment by popindavibe** [comment]
Testing with old `zsh-common` package and re-tested scenario 2
- previously installed ksh package is removed during prep
- newer zsh-common package is downloaded
- `auter --apply` works OK
```
root@xenial:/home/builder/auter# cat /var/lib/auter/last-apply-output-default
Reading package lists...
Building dependency tree...
Reading state information...
The following package was automatically installed and is no longer required:
  libfreetype6
Use 'apt autoremove' to remove it.
Suggested packages:
  zsh-doc
Recommended packages:
  zsh
The following packages will be upgraded:
  zsh-common
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 3177 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 zsh-common all 5.1.1-1ubuntu2.2 [3177 kB]
Fetched 3177 kB in 3s (1057 kB/s)
(Reading database ... 36038 files and directories currently installed.)
Preparing to unpack .../zsh-common_5.1.1-1ubuntu2.2_all.deb ...
Unpacking zsh-common (5.1.1-1ubuntu2.2) over (5.1.1-1ubuntu2) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up zsh-common (5.1.1-1ubuntu2.2) ...
```

### Logs
```
Jul 19 19:05:26 xenial root[359]: custom pre prep script ran
Jul 19 19:05:43 xenial auter[605]: INFO: Updates downloaded to /var/cache/auter/default
Jul 19 19:05:43 xenial auter[617]: INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
Jul 19 19:05:43 xenial root[619]: custom post prep script ran
Jul 19 19:07:10 xenial auter[630]: INFO: Running with: /usr/sbin/auter --apply
Jul 19 19:07:10 xenial auter[634]: INFO: Running in an interactive shell, disabling all random sleeps
Jul 19 19:07:11 xenial auter[654]: INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
Jul 19 19:07:11 xenial root[656]: custom pre apply script ran
Jul 19 19:07:11 xenial auter[660]: INFO: Applying updates
Jul 19 19:07:16 xenial auter[1053]: INFO: Updates complete. You may need to reboot for some updates to take effect
Jul 19 19:07:16 xenial auter[1057]: INFO: Auter successfully ran at 2018-07-19T19:07:16+00:00
Jul 19 19:07:16 xenial auter[1060]: INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script
Jul 19 19:07:16 xenial root[1062]: custom post apply script ran
```

**Comment by popindavibe** [comment]
# Scenario 3: Using cron, download from prep and reboot on updates

**[ OK ] PREP**
**[ KO ] APPLY**

## Testing prep via cron
- `/etc/cron.d/auter`:
```bash
#Ansible: prep auter
03 21 28 1-11 * root /usr/sbin/auter --prep
```
- Logs
```
Jul 28 21:03:01 xenial CRON[11940]: (root) CMD (/usr/sbin/auter --prep)
Jul 28 21:03:01 xenial auter[11949]: INFO: Running with: /usr/sbin/auter --prep
Jul 28 21:03:01 xenial auter[11953]: INFO: Running Pre-Prep script /etc/auter/pre-prep.d/pre_prep_script
Jul 28 21:03:01 xenial root[11955]: custom pre prep script ran

Jul 28 21:26:42 xenial auter[13318]: INFO: Updates downloaded to /var/cache/auter/default
Jul 28 21:26:42 xenial auter[13333]: INFO: Running Post-Prep script /etc/auter/post-prep.d/post_prep_script
Jul 28 21:26:42 xenial root[13335]: custom post prep script ran
```

## Testing apply via cron

- `/etc/cron.d/auter`:
```bash
#Ansible: apply auter
50 21 22-28 1-11 * root [ $(date +\%u) -eq 6 ] &&  /usr/sbin/auter --apply
```
- Logs
```
Jul 28 21:50:01 xenial auter[13490]: INFO: Running with: /usr/sbin/auter --apply
Jul 28 21:50:01 xenial auter[13586]: INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
Jul 28 21:50:01 xenial root[13588]: custom pre apply script ran

Jul 28 22:38:01 xenial auter[13693]: INFO: Applying updates
Jul 28 22:38:06 xenial auter[13836]: WARNING: No updates were applied. 16 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```
Email received:
```
From root@xenial  Sat Jul 28 22:38:06 2018
Return-Path: <root@xenial>
X-Original-To: root
Delivered-To: root@xenial
Received: by xenial (Postfix, from userid 0)
        id EA033F61A; Sat, 28 Jul 2018 22:38:06 +0100 (BST)
From: root@xenial (Cron Daemon)
To: root@xenial
Subject: Cron <root@xenial> [ $(date +%u) -eq 6 ] &&  /usr/sbin/auter --apply
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
X-Cron-Env: <SHELL=/bin/sh>
X-Cron-Env: <HOME=/root>
X-Cron-Env: <PATH=/usr/bin:/bin>
X-Cron-Env: <LOGNAME=root>
Message-Id: <20180728213806.EA033F61A@xenial>
Date: Sat, 28 Jul 2018 22:38:06 +0100 (BST)

Trying to update
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin:
E: Sub-process /usr/bin/dpkg returned an error code (2)
```

**Comment by popindavibe** [comment]
So, previous fail shows that Auter is not setting the appropriate flags to support automatic update in case of an update coming with a dialog. This is not a cron issue.

On `prepare_update` function in, I can see we use `declare -x DEBIAN_FRONTEND=noninteractive
`, however it is not reported in `apply_updates`.

Adding it at the top the the function does the trick to get Auter installing packages that would trigger a dialog box in interactive mode. This probably needs to be discussed further, adding @reaperzn to the review.

### Change 1
```
function apply_updates() {
  # Prevent dialog box as we are not running the update in interactive mode
  declare -x DEBIAN_FRONTEND=noninteractive
```
- Logs in /var/lib/auter/last-apply-output-default
```
dpkg: warning: 'ldconfig' not found in PATH or not executable
dpkg: warning: 'start-stop-daemon' not found in PATH or not executable
dpkg: error: 2 expected programs not found in PATH or not executable
Note: root's PATH should usually contain /usr/local/sbin, /usr/sbin and /sbin
```

### Change 2
I added the full PATH in `/etc/cron.d/auter` and tried again, it went through this time.
```
root@xenial:~# head /etc/cron.d/auter
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
```
- Logs OK
```
Jul 29 14:17:01 xenial CRON[15169]: (root) CMD ([ $(date +%u) -eq 7 ] && /usr/sbin/auter --apply )
Jul 29 14:17:01 xenial auter: INFO: Running with: /usr/sbin/auter --apply
Jul 29 14:17:02 xenial auter: INFO: Running Pre-Apply script /etc/auter/pre-apply.d/pre_apply_script
Jul 29 14:17:05 xenial auter: INFO: Applying updates
Jul 29 14:17:26 xenial auter: INFO: Updates complete. You may need to reboot for some updates to take effect
Jul 29 14:17:26 xenial auter: INFO: Auter successfully ran at 2018-07-29T15:17:26+01:00
Jul 29 14:17:26 xenial auter: INFO: Running Post-Apply script /etc/auter/post-apply.d/pre_apply_script
```
```
root@xenial:~# tail /var/lib/auter/last-apply-output-default
Installing new version of config file /etc/profile.d/apps-bin-path.sh ...
snapd.snap-repair.service is a disabled or a static unit, not starting it.
Setting up ubuntu-core-launcher (2.34.2) ...
Setting up libslang2:amd64 (2.3.0-2ubuntu1.1) ...
Setting up libglib2.0-0:amd64 (2.48.2-0ubuntu3) ...
No schema files found: doing nothing.
Setting up libglib2.0-data (2.48.2-0ubuntu3) ...
Setting up shared-mime-info (1.5-2ubuntu0.2) ...
Setting up linux-libc-dev:amd64 (4.4.0-131.157) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
```

**Comment by popindavibe** [comment]
Note: unattended-upgrade would typically run through cron.daily, which sets full PATH (/etc/crontab) and so does not have this problem.
Depending what are current requirements for Auter compatibility on Ubuntu, this may or may not be something that needs to be fixed for 0.12 release.

**Comment by popindavibe** [comment]
Re-tested last scenario and tested cron+frompreponly+reboot => **OK**
```
Aug  1 17:00:01 xenial auter: INFO: Running with: /usr/sbin/auter --prep
Aug  1 17:00:28 xenial auter: INFO: Updates downloaded to /var/cache/auter/default
Aug  1 17:05:01 xenial auter: INFO: Running with: /usr/sbin/auter --apply
Aug  1 17:05:22 xenial auter: INFO: Applying updates
Aug  1 17:07:48 xenial auter: INFO: Updates complete. You may need to reboot for some updates to take effect
Aug  1 17:07:48 xenial auter: INFO: Auter successfully ran at 2018-08-01T18:07:48+01:00
Aug  1 17:07:48 xenial auter: INFO: Rebooting server
```
Closing.
